src/Framework/ExecutionOrderDependency.php

```diff
@@ -0,0 +1,184 @@
+<?php declare(strict_types=1);
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace PHPUnit\Framework;
+
+use function array_filter;
+use function array_map;
+use function array_search;
+use function count;
+use function explode;
+use function strpos;
+use function trim;
+
+/**
+ * @internal This class is not covered by the backward compatibility promise for PHPUnit
+ */
+final class ExecutionOrderDependency
+{
+    /* var string */
+    private $className = '';
+
+    /* var string */
+    private $methodName = '';
+
+    /* var boolean */
+    private $useShallowClone = false;
+
+    /* var boolean */
+    private $useDeepClone = false;
+
+    public static function createFromDependsAnnotation(string $className, string $annotation): self
+    {
+        // Split clone option and target
+        $parts = explode(' ', trim($annotation), 2);
+
+        if (count($parts) === 1) {
+            $cloneOption = '';
+            $target      = $parts[0];
+        } else {
+            $cloneOption = $parts[0];
+            $target      = $parts[1];
+        }
+
+        // Prefix provided class for targets assumed to be in scope
+        if ($target !== '' && strpos($target, '::') === false) {
+            $target = $className . '::' . $target;
+        }
+
+        return new self($target, null, $cloneOption);
+    }
+
+    /**
+     * @param array<ExecutionOrderDependency> $dependencies
+     *
+     * @return array<ExecutionOrderDependency>
+     */
+    public static function filterInvalid(array $dependencies): array
+    {
+        return array_filter($dependencies, function (self $d) {
+            return $d->isValid();
+        });
+    }
+
+    /**
+     * @param array<ExecutionOrderDependency> $existing
+     * @param array<ExecutionOrderDependency  $additional
+     *
+     * @return array<ExecutionOrderDependency>
+     */
+    public static function mergeUnique(array $existing, array $additional): array
+    {
+        $existingTargets = array_map(function ($dependency) {
+            return $dependency->getTarget();
+        }, $existing);
+
+        foreach ($additional as $dependency) {
+            if (array_search($dependency->getTarget(), $existingTargets, true) === false) {
+                $existingTargets[]  = $dependency->getTarget();
+                $existing[]         = $dependency;
+            }
+        }
+
+        return $existing;
+    }
+
+    /**
+     * @param array<ExecutionOrderDependency> $left
```

Comment:
should probably be `@param ExecutionOrderDependency[] $left`; but this may work too. Just don't know for sure :)

Reply:
This works too and I don't mind either way. The `array<...>` notation is more flexible for more complex data structures.